osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -232,7 +233,7 @@ public void Filter(FilterCriteria newCriteria = null, bool debounce = true)
             if (newCriteria != null)
                 criteria = newCriteria;
 
-            if (!IsLoaded) return;
+            if (LoadState != LoadState.Alive) return;

```

Comment:
I think I'd prefer if we fixed `IsLoaded` rather than temporarily introducing noise via `LoadState != LoadState.Alive` that has to be cleaned up later on anyway.